packages/menu/src/menu.tsx

```diff
@@ -141,7 +141,11 @@ const motionVariants: Variants = {
   },
 }
 
-const Motion = chakra(motion.div)
+// @future: only call `motion(chakra.div)` when we drop framer-motion v3 support
+const MotionDiv: CustomDomComponent<PropsOf<typeof chakra.div>> =
+  "custom" in motion
+    ? (motion as any).custom(chakra.div)
+    : (motion as any)(chakra.div)
```

Comment:
I think this change is not necessary, because it used the chakra factory, not `motion.custom`.

`chakra(motion.div)` and `motion(chakra.div)` are working differently, specifically with the `transition` prop.

afaik `chakra(motion.div)` does not pass `transition` on to the motion component, where as `motion(chakra.div)` does not pass transition to the chakra component.

Reply:
Hmmmm, you might be right. This was the only generated component where I was receiving type errors, I think specifically for the types of `onDrag`. I'll re-investigate and see what needs done. Wanna mark as changes requested so this isn't merged till we're sure?

Reply:
I think those type errors are not related to the framer-motion update. The errors occurred with removing the any cast in this commit: https://github.com/chakra-ui/chakra-ui/commit/efd32c982cd58c1e53468743346a3b8f482b1ae3

I think your change is actually a viable solution for that, so I won't mark this as changes requested.
IMO this is good to go!

Reply:
Awesome! Want to give it an approval instead then? 😁 Or should we just merge it? I feel like this is fairly important due to the downstream issues it's causing.

Reply:
I think this change is needed.